src/BTVNanoCommissioning/workflows/ctag_Wc_valid_sf.py

```diff
@@ -533,296 +379,259 @@ def process(self, events):
+                    "btagDeepFlavCvB": jetsfs_c,
                 }
-                if isRealData:
-                    h.fill(dataset=dataset, **fields, weight=osss)
-                else:
-                    h.fill(
-                        dataset=dataset,
-                        **fields,
-                        weight=weights.weight()[event_level] * osss,
-                    )
-            elif ["zmass", "zpt", "zeta", "zphi"] == histname:
-                fields = {l: sz[l] for l in h.fields if l in dir(sz)}
-                if isRealData:
-                    h.fill(dataset=dataset, char=osss, **fields, weight=osss)
-                else:
-                    h.fill(
-                        dataset=dataset,
-                        char=osss,
-                        **fields,
-                        weight=weights.weight()[event_level] * osss,
-                    )
-            elif ["wmass", "wpt", "weta", "wphi"] == histname:
-                fields = {l: sw[l] for l in h.fields if l in dir(sw)}
-                if isRealData:
-                    h.fill(dataset=dataset, char=osss, **fields, weight=osss)
-                else:
-                    h.fill(
-                        dataset=dataset,
-                        char=osss,
-                        **fields,
-                        weight=weights.weight()[event_level] * osss,
-                    )
+
+        for histname, h in output.items():
+            if "Deep" in histname and "btag" not in histname:
+                h.fill(
+                    flatten(genflavor),
+                    flatten(sjets[histname]),
+                    weight=flatten(
+                        ak.broadcast_arrays(
+                            weights.weight()[event_level] * osss, sjets["pt"]
+                        )[0]
+                    ),
+                )
+            elif "jet_" in histname and "mu" not in histname:
+                h.fill(
+                    flatten(genflavor),
+                    flatten(sjets[histname.replace("jet_", "")]),
+                    weight=flatten(
+                        ak.broadcast_arrays(
+                            weights.weight()[event_level] * osss, sjets["pt"]
+                        )[0]
+                    ),
+                )
+            elif "hl_" in histname and histname.replace("hl_", "") in shmu.fields:
+                h.fill(
+                    flatten(shmu[histname.replace("hl_", "")]),
+                    weight=weights.weight()[event_level] * osss,
+                )
+            elif (
+                "soft_l" in histname and histname.replace("soft_l_", "") in ssmu.fields
+            ):
+                h.fill(
+                    flatten(ssmu[histname.replace("soft_l_", "")]),
+                    weight=weights.weight()[event_level] * osss,
+                )
+            elif "MET" in histname:
+                h.fill(
+                    flatten(selev.METFixEE2017[histname.replace("MET_", "")]),
```

Comment:
this will not work for 2016 and 2018 samples, as branch name is specific for 2017